### ISSUE TITLE
filename and timestamps storing

### ISSUE THREAD

**Issue by cieplutki** [ebiggers/libdeflate#30]
Hey,
is gzip able to save file name or time stams of compressed file?
I looked into options but there seem to be nothing which suggests there is no such option.

**Comment by ebiggers** [comment]
No, storing those fields is not implemented yet.  I think they are rarely needed, actually.  Why do you need them?

**Comment by cieplutki** [comment]
I use it as archiver not only compressor and so I don't have to do this: 
```
$ tar c file.name | gzip - > file.name.tar.gz
(...)
$ tar xf file.name.gz | gunzip
```

Sometimes I have to deal with damaged/recovered archives and filename at that point does make a difference. If you have yet another "anonymous" file that god one knows whats inside - there is a chance it will be "ignored" on purpose where there are tens or hundreds of similar ones recovered from file system crash. In my experience the most frequent fs crashes "produce" files that are "cut off" at the end. Whenever you try to undelete or recover raw files (without filesystem - name/directory structure) they are somewhat incomplete at the end (due to file fragmentation) and then name of file can give you a clue whether it's what you are looking for or not.

PS. Sorry for lengthy explanation, hope it is not off-topic.

**Comment by Dobatymo** [comment]
The name gzip suggests a gzip alternative... The original software stores filename and modtime by default. So I wouldn't say it's a rarely needed feature...
